android/quest/src/main/java/org/smartregister/fhircore/quest/ui/report/measure/MeasureReportViewModel.kt

```diff
@@ -360,164 +327,188 @@ constructor(
     measureUrl: String,
     startDateFormatted: String,
     endDateFormatted: String,
-    indicatorTitle: String
-  ) {
-    val measureReport: MeasureReport? =
-      withContext(dispatcherProvider.io()) {
-        try {
-          if (measureUrl.contains("Stock")) {
-            fhirEngine
-              .search<Group> {
-                filter(
-                  Group.TYPE,
-                  {
-                    value =
-                      of(
-                        Coding(
-                          GroupType.MEDICATION.system,
-                          GroupType.MEDICATION.toCode(),
-                          GroupType.MEDICATION.display
-                        )
-                      )
-                  }
-                )
-              }
-              .map {
-                fhirOperator.evaluateMeasure(
-                  measureUrl = measureUrl,
-                  start = startDateFormatted,
-                  end = endDateFormatted,
-                  reportType = POPULATION,
-                  subject = it.id,
-                  practitioner = null
-                  /* TODO DO NOT pass this id to MeasureProcessor as this is treated as subject if subject is null.
-                  practitionerId?.asReference(ResourceType.Practitioner)?.reference*/ ,
-                  lastReceivedOn = null // Non-null value not supported yet
-                )
-              }
-              .first()
-          } else
-            fhirOperator.evaluateMeasure(
-              measureUrl = measureUrl,
-              start = startDateFormatted,
-              end = endDateFormatted,
-              reportType = POPULATION,
-              subject = null,
-              practitioner = null
-              /* TODO DO NOT pass this id to MeasureProcessor as this is treated as subject if subject is null.
-              practitionerId?.asReference(ResourceType.Practitioner)?.reference*/ ,
-              lastReceivedOn = null // Non-null value not supported yet
-            )
-        } catch (exception: IllegalArgumentException) {
-          Timber.e(exception)
-          null
+    subjectXFhirQuery: String?
+  ): List<MeasureReport> {
+    val measureReport = mutableListOf<MeasureReport>()
+    withContext(dispatcherProvider.io()) {
+      if (subjectXFhirQuery?.isNotEmpty() == true) {
+        fhirEngine
+          .search(subjectXFhirQuery)
+          .map {
+            // TODO a hack to prevent missing subject in case of Group based reports where
+            // MeasureEvaluator looks for Group members and skips the Group itself
```

Comment:
Does this happen because of the way a measure is written, or does evaluate-measure always look for a Group member whenever you pass it a Group?

Reply:
Yeh. The measure evaluator ignores the given Group subject and just looks for embedded entities